.github/workflows/ci.yml

```diff
@@ -0,0 +1,59 @@
+name: CI workflow
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+        
+    - name: Setup node
+      uses: actions/setup-node@v2
+      with:
+        node-version: 14.x
+        
+    - name: Install dependencies
+      run: npm ci
+      env:
+        CI: true
+    
+    - name: Cache npm packages
+      id: npm-cache
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-npm-
+        
+    - name: Lint files
+      run: npm run lint
+
+    - name: Linting failed (attempting fix...)
+      if: ${{ failure() }}
+      run: npm run lint:fix
+
+    - name: Commit fixed lint issues
+      uses: stefanzweifel/git-auto-commit-action@v4.1.2
+      with:
+        commit_message: Apply fixed lint issues
+        branch: ${{ github.head_ref }}
+
+    - name: Run test suite
```

Comment:
I would prefer the testing to be done in a separate job that is not blocked by lint issues.

Reply:
The CI flow should be blocked if code has lint issues and an automatic fix attempt fails. If it is not lintable, it is not testable after all. If you wish to propose a change, please update the PR 😄

Reply:
I have updated the CI configuration and moved the testing to a separate job that will run in parallel with the linting. Please see the changes in the last commit. Regarding the non lintable code is also not testable, I'm not sure about that - linting is about style and may not apply, but still preferable, and testing is about functionality that works as expected. For such a small library it's not critical to have these processes step by step, but for a long-running test suite, it's definitely better to have it in parallel.